File: ForegroundAppWithBackgroundApp/CS/MyBackgroundApplication/StartupTask.cs
using System;
using Windows.ApplicationModel.Background;
using Windows.Devices.Gpio;
using Windows.System.Threading;

// The Background Application template is documented at http://go.microsoft.com/fwlink/?LinkID=533884&clcid=0x409

namespace MyBackgroundApplication
{
    public sealed class StartupTask : IBackgroundTask
    {
        BackgroundTaskDeferral deferral;
        ThreadPoolTimer timer;
        GpioPin pin;
        GpioPinValue pinValue;
        const int LED_PIN = 6;

        public void Run(IBackgroundTaskInstance taskInstance)
        {
            deferral = taskInstance.GetDeferral();
            InitGPIO();
            if (pin != null)
            {
                timer = ThreadPoolTimer.CreatePeriodicTimer(new TimerElapsedHandler(Timer_Tick), TimeSpan.FromMilliseconds(500));
            }
        }

        private void Timer_Tick(ThreadPoolTimer timer)
        {
            if (pinValue == GpioPinValue.High)
            {
                pinValue = GpioPinValue.Low;
                pin.Write(pinValue);
            }
            else
            {
                pinValue = GpioPinValue.High;
                pin.Write(pinValue);
            }
        }


        private void InitGPIO()
        {
            var gpio = GpioController.GetDefault();

            // Show an error if there is no GPIO controller
            if (gpio == null)
            {
                pin = null;
                return;
            }

            pin = gpio.OpenPin(LED_PIN);
            pinValue = GpioPinValue.High;
            pin.Write(pinValue);
            pin.SetDriveMode(GpioPinDriveMode.Output);
        }
    }
}

File: ForegroundAppWithBackgroundApp/CPP/MyBackgroundApplicationCpp/StartupTask.cpp
#include "pch.h"
#include "StartupTask.h"

using namespace MyBackgroundApplicationCpp;

using namespace Platform;
using namespace Windows::ApplicationModel::Background;
using namespace Windows::Foundation;

// The Background Application template is documented at http://go.microsoft.com/fwlink/?LinkID=533884&clcid=0x409

const int LED_PIN = 5;
const int ticksPerMillisecond = 10000;

void StartupTask::Run(IBackgroundTaskInstance^ taskInstance)
{
    deferral = taskInstance->GetDeferral();
    InitGPIO();
    if (pin != nullptr)
    {
        TimeSpan ts;
        ts.Duration = 500 * ticksPerMillisecond;
        timer = ThreadPoolTimer::CreatePeriodicTimer(ref new TimerElapsedHandler(this, &StartupTask::Timer_Tick), ts);
    }
}

void StartupTask::Timer_Tick(ThreadPoolTimer^ timer)
{
    if (pinValue == GpioPinValue::High)
    {
        pinValue = GpioPinValue::Low;
        pin->Write(pinValue);
    }
    else
    {
        pinValue = GpioPinValue::High;
        pin->Write(pinValue);
    }
}


void StartupTask::InitGPIO()
{
    auto gpio = GpioController::GetDefault();

    // Show an error if there is no GPIO controller
    if (gpio == nullptr)
    {
        pin = nullptr;
        return;
    }

    pin = gpio->OpenPin(LED_PIN);
    pinValue = GpioPinValue::High;
    pin->Write(pinValue);
    pin->SetDriveMode(GpioPinDriveMode::Output);
}